app/src/main/java/com/termux/x11/XrActivity.java

```diff
@@ -183,8 +186,28 @@ public void onDrawFrame(GL10 gl10) {
         }
     }
 
+    @Override
+    public void onWindowFocusChanged(boolean hasFocus) {
+        super.onWindowFocusChanged(hasFocus);
+        this.hasFocus = hasFocus;
+        if (hasFocus) {
+            keyboardDisabled = true;
+            new Handler().postDelayed(() -> keyboardDisabled = false, 250);
+        }
+    }
+
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
```

Comment:
Does `LorieView`'s dispatchKeyEventPreIme work in the case if Activity's dispatchKeyEvent is overridden?

Reply:
Software keyboard sends randomly event with KEYCODE_MEDIA_PLAY into dispatchKeyEventPreIme.
Hardware keyboard works sends correct events into dispatchKeyEventPreIme.

Reply:
I am sorry, I do not understand how is your question related to the code.

What change do you require here?